util/validate2baseline

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+$LOAD_PATH.unshift File.expand_path("../lib", __dir__)
+
+require "optparse"
+require "cli"
+
+DEFAULT_TEST_TYPE = "apples to apples"
+DESCRIPTION = <<-DESC
+  `validate2baseline` validates performance results data against baseline data.
+
+  You must specify a --baseline for the baseline data and a --results_dir
+  containing the performance results to compare for validation.
+
+  The --test_type is optional and defaults to "#{DEFAULT_TEST_TYPE}"
+
+  A passing state will echo "PASS" to STDOUT and exit with 0. A failing state
+  will echo "FAIL" to STDOUT and exit with 1.
+
+DESC
+
+DEFAULTS = <<~DEFAULTS
+  The following defaults values are used if the options are not specified:
+  * TEST_TYPE (-t, --test_type): #{DEFAULT_TEST_TYPE}
+DEFAULTS
+
+options = {}
+optparse = OptionParser.new do |opts|
+  options[:test_type] = DEFAULT_TEST_TYPE
+  opts.on("-t", "--test_type TEST_TYPE", String) do |o|
+    options[:test_type] = o
+  end
+  opts.on("-b", "--baseline BASELINE", String) do |o|
+    options[:baseline] = o
+  end
+  opts.on("-r", "--results_dir DIR", String) do |o|
+    options[:results_dir] = o
+  end
+  opts.on("-h", "--help") do
+    puts DESCRIPTION
+    puts opts
+    puts DEFAULTS
+    exit
+  end
+end
+
+# verify mandatory options have been set
+begin
+  optparse.parse!
+  mandatory = %i[test_type baseline results_dir]
+  missing = mandatory.select { |o| options[o].nil? }
+  raise OptionParser::MissingArgument, missing.join(", ") unless missing.empty?
+rescue OptionParser::InvalidOption, OptionParser::MissingArgument
+  puts $ERROR_INFO.to_s
+  puts optparse
+  exit
+end
+
+c = GPLT::CLI.new
```

Comment:
Now that this is a standalone script what is the benefit of going through `CLI` rather than just calling `validate_results_to_baseline` directly? Not saying there isn't a benefit, just want to understand it :)

Reply:
I have the same question. :)

Reply:
A class instance is required for the helpers in the PerfRunHelper module to work as expected.  Normally this happens in the context of a Beaker instance, but here a class wrapper needs to be created to satisfy that requirement.

Reply:
I was able to avoid using `cli.rb` by updating `validate2baseline` with the following:
* require `perf_run_helper` and include `PerfRunHelper`
```
# require "cli"

require "./tests/helpers/perf_run_helper.rb"
include PerfRunHelper # rubocop:disable Style/MixinUsage
```
* call `validate_results_to_baseline` directly:
```
# c = GPLT::CLI.new
# c.validate2baseline(options)

validate_results_to_baseline(options[:results_dir], options[:baseline], options[:test_type])
```

This is how the other util scripts do it and seems to work fine unless I'm not understanding some aspect of this (which is entirely possible).

Reply:
Without bringing in the conditional logic from `cli.rb` the interface contract for the application is broken.  It won't emit the PASS/FAIL status and it won't return the expected exit code on failure.
Stylistically, I would wrap it in class anyway to avoid having all methods in the Object rather than namespaced (thus avoiding having to squash rubocop).
Having said that, I am not against removing the `cli.rb` file and bringing that code into the `validate2baseline` file if that satisfies the criteria of being more consistent with the other `util` items.

Reply:
Sorry for overlooking the conditional logic; I was just trying to see if I could get the perf_results_helper functionality working within the script without relying on `cli.rb`. In addition to being consistent with the other scripts, having less 'parts' makes it easier to understand (for me).

Reply:
I have also just complicated the issues by adding a `--log-level` flag which needs to influence the object outside of the context of the `validate_results_to_baseline`.  I believe this to be assisted by current structure as well.  Feel free to take a look and see if your opinion differs.  Thanks Bill.